stdlib/asyncio/base_futures.pyi

```diff
@@ -11,7 +11,7 @@ _PENDING: Literal["PENDING"]  # undocumented
 _CANCELLED: Literal["CANCELLED"]  # undocumented
 _FINISHED: Literal["FINISHED"]  # undocumented
 
-def isfuture(obj: object) -> bool: ...
+def isfuture(obj: object) -> TypeGuard[futures.Future]: ...
```

Comment:
At runtime this allows any object with a specific attribute:
```
    return (hasattr(obj.__class__, '_asyncio_future_blocking') and
            obj._asyncio_future_blocking is not None)
```
I don't think that's enough for a TypeGuard.

Reply:
The docs put it this way:

https://docs.python.org/3/library/asyncio-future.html#asyncio.isfuture
> a Future-like object with a _asyncio_future_blocking attribute.

https://github.com/python/cpython/blob/c27a33132be101e246ae2584f1826477357138d6/Lib/asyncio/futures.py#L60-L63
>     # - Its presence is a marker to declare that a class implements
>     #   the Future protocol (i.e. is intended to be duck-type compatible).
>     #   The value must also be not-None, to enable a subclass to declare
>     #   that it is not compatible by setting this to None.

In my opinion, this is enough to be a `TypeGuard`

Reply:
The doc regarding `_asyncio_future_blocking` is correct.

I have another question. `asyncio.Future` is a generic type.
Should the typeguard reflect it, annotating at least `Future[Any]`?

Reply:
Done! Thanks!